### PR TITLE
fix: Seeds now drop and create instead of truncating

### PR DESCRIPTION
**Pull request by ericf-firebolt**
Resolves https://packboard.atlassian.net/browse/FIR-10985

### Description
`dbt seed` would not run correctly because Firebolt doesn't support truncate. Tables are now `DROP CASCADE`ed and then `CREATE`ed.

### Checklist

- [x] I have run this code in development and it appears to resolve the stated issue.
- [x] This PR includes tests, or tests are not required/relevant for this PR.
- [x] I have updated `CHANGELOG.md` and added information about my change.
- [x] If this PR requires a new PyPI release I have bumped the version number.
- [x] I have pulled/merged from the main branch if there are merge conflicts.
- [x] I have verified that this PR contains only code changes relevant to this PR.

### CODE REVIEW

File: dbt/include/firebolt/macros/adapters.sql [hunk 1]
```diff
@@ -178,15 +178,23 @@
...
+   Get table list from SHOW TABLES and save. After drop, recreate table. #}
+    {% call statement('list_tables', fetch_result=True) -%}
+        SHOW TABLES
+    {%- endcall %}
+    {% set tables = load_result('list_tables')['data'] %}
+    {% call statement('table_schema') -%}
+        DROP TABLE {{ relation.identifier }} CASCADE
+    {%- endcall %}
+    {# SHOW TABLES returns a dictionary. We want the list of tuples at key=='data'.
+       Unluckily, we have to step through each one to find the one that corresponds
```
Comment (miguel-firebolt) [inline dbt/include/firebolt/macros/adapters.sql:191]: Can you use agate for this? I think I saw it used in other places

Comment (ericf-firebolt) [reply dbt/include/firebolt/macros/adapters.sql:191]: You mean to find the right table, instead of using the loop?

Comment (miguel-firebolt) [reply dbt/include/firebolt/macros/adapters.sql:191]: Yes -- I think there are examples in the core repo where they use agate to filter

Comment (ericf-firebolt) [reply dbt/include/firebolt/macros/adapters.sql]: Took me a while, but the example was actually in our adapter code. Anyway, done now.

**Review by octavianzarzu30 — APPROVED** [review]

**Review by stepansergeevitch — APPROVED** [review]
Please make sure to run integration and jaffle shop tests before merging

**Pull request merged**